src/module/util/misc.ts

```diff
@@ -132,3 +137,40 @@ export async function tokenScrollText(
   // @ts-expect-error v11 types
   await canvas.interface.createScrollingText(token.center, content, style);
 }
+
+//Used to check actor may use talent
+//Perhaps should be LancerPILOT method
+export function isTalentAvailable(actor: LancerActor | undefined, talentSlug: string): boolean {
+  if (!actor?.is_mech() || !actor.system.pilot?.value?.is_pilot()) return false;
+
+  let talents = actor.system.pilot.value.items.filter(i => i.is_talent());
+
+  //Go through the slugs of all the available talent ranks
+  for (const talent of talents) {
+    let rank_num = talent.system!.curr_rank;
+    for (let i = 0; i < rank_num; i++) {
+      const rank_name = talent.system!.ranks[i].name;
+      if (slugify(rank_name, "-") === talentSlug) return true;
+    }
+  }
+
+  return false;
+}
```

Comment:
I'd suggest using the talent's `system.lid` and rank number as the arguments here, instead of slugifying the rank's name. The current approach will break as soon as someone localizes or re-fluffs a talent, or if they add a homebrew talent which happens to reuse another talent's rank title.